libs/marshalling/algebra/CMakeLists.txt

```diff
@@ -14,15 +14,6 @@ include(CMDeploy)
 include(CMSetupVersion)
 
 cm_workspace(marshalling)
-
-macro(cm_find_package NAME)
-    if(NOT "${NAME}" MATCHES "^${CMAKE_WORKSPACE_NAME}_.*$" AND NOT "${NAME}" STREQUAL CM)
-        find_package(${ARGV})
-    else()
-        set(${ARGV0}_FOUND ON CACHE BOOL "")
-    endif()
-endmacro()
-
```

Comment:
Since the changes affect CMake, I have overall comments on this.

Take a look at [this commit](https://github.com/NilFoundation/crypto3-algebra/commit/f4a68ed9de9bc47eadf165be17c7172e33edc11b). You should note:
- `${Boost_LIBRARIES}` -> `Boost::lib_name`
- Removal of `if(ENABLE_JUNIT_TEST_OUTPUT)`. We now have flags to mimic such behavior, they are in flake.nix
- Removal of `get_target_property(target_type Boost::unit_test_framework TYPE)`. It automatically substitutes these flags
- Removal of `if(NOT Boost_FOUND AND NOT CMAKE_CROSSCOMPILING)`. It's just nonsensical part.

In addition, you can call find_package(Boost REQUIRED COMPONENTS ...) once at the root of crypto3 and remove find_package from the submodules entirely.

Reply:
Ofc, you should move this into different task. But at least adjust changes in this PR according to this list

Reply:
cmake improvements moved here
https://github.com/NilFoundation/crypto3/pull/233